ngraph/core/reference/include/ngraph/runtime/reference/interpolate.hpp

```diff
@@ -0,0 +1,522 @@
+                        {
+                            return static_cast<int64_t>(std::floor(x_original));
+                        }
+                        return static_cast<int64_t>(std::round(x_original));
+                    };
+                }
+            };
+
+            class GetOriginalCoordinate final
+            {
+            public:
+                GetOriginalCoordinate()
+                    : GetOriginalCoordinate(Transform_mode::half_pixel)
+                {
+                }
+
+                GetOriginalCoordinate(Transform_mode mode)
+                    : m_mode{mode}
+                    , m_func{get_func(mode)}
+                {
+                }
+
+                ~GetOriginalCoordinate() = default;
+
+                float operator()(float x_resized,
+                                 float x_scale,
+                                 float length_resized,
+                                 float length_original) const
+                {
+                    return m_func(x_resized, x_scale, length_resized, length_original);
+                }
+
+            private:
+                using Func = std::function<float(float, float, float, float)>;
+
+                Transform_mode m_mode;
+                Func m_func;
+
+                Func get_func(Transform_mode mode)
+                {
+                    switch (mode)
+                    {
+                    case Transform_mode::pytorch_half_pixel:
+                        return [](float x_resized, float x_scale, float length_resized, float) {
+                            return length_resized > 1 ? (x_resized + 0.5f) / x_scale - 0.5f : 0.0f;
+                        };
+                        break;
+                    case Transform_mode::asymmetric:
+                        return [](float x_resized, float x_scale, float, float) {
+                            return x_resized / x_scale;
+                        };
+                        break;
+                    case Transform_mode::tf_half_pixel_for_nn:
+                        return [](float x_resized, float x_scale, float, float) {
+                            return (x_resized + 0.5f) / x_scale;
+                        };
+                        break;
+                    case Transform_mode::align_corners:
+                        return [](
+                            float x_resized, float, float length_resized, float length_original) {
+                            return length_resized == 1
+                                       ? 0
+                                       : x_resized * (length_original - 1) / (length_resized - 1);
+                        };
+                        break;
+                    default:;
+                    }
+                    return [](float x_resized, float x_scale, float, float) {
+                        return ((x_resized + 0.5f) / x_scale) - 0.5f;
+                    };
+                }
+            };
```

Comment:
If "x_scale == 1.f" or "length_resized  == length_original", we need return "x_resized" directly. 
Take an example, when dataShape is [B, C H W] == [4, 4, 10, 10], targetShape is [4, 4, 30, 30], mode is "nearest", transform_mode is "tf_half_pixel_for_nn", round mode is "ceil". For first dimension(batch), the first batch's transformed position is (0 + 0.5) / 1.f = 0.5, after round by ceil , it's 1(second batch). That means we set data to first batch with data from second batch. This is not reasonable. I think the coordinate transformation is only suitable when scales is not 1. what do you think?

Reply:
I agree. Fixed in `operator ()` of this class.